grafanalib/tests/test_parsing.py

```diff
@@ -0,0 +1,444 @@
+
+
+def datasources():
+    return st.builds(G.DataSourceInput,
+                     name=unknown(),
+                     label=unknown(),
+                     pluginId=unknown(),
+                     pluginName=unknown(),
+                     description=unknown())
+
+
+def constantinputs():
+    return st.builds(G.ConstantInput,
+                     name=unknown(),
+                     label=unknown(),
+                     value=unknown(),
+                     description=unknown())
+
+
+def inputs():
+    return datasources() | constantinputs()
+
+
+def annotations():
+    return st.builds(G.Annotations,
+                     list=st.lists(max_size=0))
+
+
+def templating():
+    return st.builds(G.Templating,
+                     list=st.lists(max_size=0))
+
+
+def templates():
+    return st.builds(G.Template,
+                     default=unknown(),
+                     dataSource=unknown(),
+                     label=unknown(),
+                     name=unknown(),
+                     query=unknown(),
+                     allValue=unknown(),
+                     includeAll=st.booleans(),
+                     multi=st.booleans(),
+                     regex=unknown())
+
+
+def times():
+    return st.builds(G.Time,
+                     start=unknown(),
+                     end=unknown())
+
+
+def timepickers():
+    return st.builds(G.TimePicker,
+                     refreshIntervals=unknown(),
+                     timeOptions=unknown(),
+                     collapse=st.booleans(),
+                     enable=unknown(),
+                     notice=unknown(),
+                     now=unknown(),
+                     status=unknown())
+
+
+def dashboards():
+    return st.builds(G.Dashboard,
+                     title=st.text(string.printable),
+                     rows=st.lists(rows()),
+                     annotations=annotations(),
+                     editable=st.booleans(),
+                     gnetId=unknown(),
+                     hideControls=st.booleans(),
+                     id=unknown(),
+                     inputs=st.lists(inputs()),
+                     links=st.lists(max_size=0),
+                     refresh=unknown(),
+                     schemaVersion=st.integers(),
+                     sharedCrosshair=st.booleans(),
+                     style=unknown(),
+                     tags=st.lists(max_size=0),
+                     templating=templating(),
+                     time=times(),
+                     timePicker=timepickers(),
+                     timezone=unknown(),
+                     version=st.integers(),
+                     graphTooltip=st.integers())
+
+
+def json_round_trip(obj):
+    """Dumps the object to a JSON and load it back. This is necessary as the
+    DashboardEncoder takes care of nested structures """
+    json_data = io.StringIO()
+    _gen.write_dashboard(obj, json_data)
+    json_data.seek(0)
+    return json.load(json_data)
```

Comment:
Another approach would be:

```python
def json_round_trip(obj):
    output = json.dumps(dashboard.to_json_data(), cls=DashboardEncoder)
    return json.loads(output)
```

Reply:
Yes much simpler ;-)
I just wanted to go trough code that already existed for some reason.